pkg/operator/configobservation/apiserver/observer_apiserver.go

```diff
@@ -0,0 +1,262 @@
+				panic(err) // programmer error, i.e. sync controller initialized with wrong namespaces
+			}
+		}
+	}()
+
+	// extract the previously observed config
+	previouslyObservedConfig := map[string]interface{}{}
+	if err := copyPreviousConfigFromExistingConfig(previouslyObservedConfig, existingConfig, certFilePath...); err != nil {
+		errs = append(errs, err)
+	}
+	if err := copyPreviousConfigFromExistingConfig(previouslyObservedConfig, existingConfig, keyFilePath...); err != nil {
+		errs = append(errs, err)
+	}
+
+	// observe the new config
+	observedConfig := map[string]interface{}{}
+	apiServer, err := listers.APIServerLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		glog.Warningf("apiservers.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	}
+	if err != nil {
+		skipResourceSync = true
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	servingCertSecretName := apiServer.Spec.ServingCerts.DefaultServingCertificate.Name
+	if len(servingCertSecretName) == 0 {
+		return observedConfig, nil
+	}
+	certFile := fmt.Sprint(userServingCertPublicCertFile)
+	if err := unstructured.SetNestedField(observedConfig, certFile, certFilePath...); err != nil {
+		skipResourceSync = true
+		return previouslyObservedConfig, append(errs, err)
+	}
+	keyFile := fmt.Sprint(userServingCertPrivateKeyFile)
+	if err := unstructured.SetNestedField(observedConfig, keyFile, keyFilePath...); err != nil {
+		skipResourceSync = true
+		return previouslyObservedConfig, append(errs, err)
+	}
+
+	sourceLocation = resourcesynccontroller.ResourceLocation{
+		Name:      servingCertSecretName,
+		Namespace: operatorclient.GlobalUserSpecifiedConfigNamespace,
+	}
+
+	return observedConfig, errs
+}
+
+func ObserveNamedCertificates(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+	listers := genericListers.(configobservation.Listers)
+	var errs []error
+	namedCertificatesPath := []string{"servingInfo", "namedCertificates"}
+
+	const (
+		maxUserNamedCerts                   = 10
+		userServingCertFormat               = "user-serving-cert-%03d"
+		userServingCertPublicCertFileFormat = "/etc/kubernetes/static-pod-resources/secrets/%v/tls.crt"
+		userServingCertPrivateKeyFileFormat = "/etc/kubernetes/static-pod-resources/secrets/%v/tls.key"
+	)
+
+	// initialize sync rules such that all named user serving certs are deleted
+	// zero-valued source location will cause destination to be deleted
+	resourceSyncRules := map[resourcesynccontroller.ResourceLocation]resourcesynccontroller.ResourceLocation{}
+	for i := 0; i < maxUserNamedCerts; i++ {
+		destination := resourcesynccontroller.ResourceLocation{Namespace: operatorclient.OperatorNamespace, Name: fmt.Sprintf(userServingCertFormat, i)}
+		resourceSyncRules[destination] = resourcesynccontroller.ResourceLocation{}
+	}
+	skipSyncResources := false
+	defer func() {
```

Comment:
Should we just do this after we exit the loop below?  If you update the comment with some indication like you have in your github comments about this being an all or nothing choice based on iterating and setting a new config, I get it.